### ISSUE TITLE
buildCheckFunction default added only to req.body

### ISSUE THREAD

**Issue by shizpi** [express-validator/express-validator#1010]
## Describe the bug

Default value is always added to `req.body` when using buildCheckFunction to validate body and query.

## To Reproduce

```
const checkBodyAndQuery = buildCheckFunction(['body', 'query']);

export const validateInvoicesParams = [
  checkBodyAndQuery('id').optional().isNumeric(),
  checkBodyAndQuery('amount').optional().custom(isRangeArray),
  checkBodyAndQuery('starts-at').default(DEFAULT_STARTS_AT).toDate().isISO8601(),
  checkBodyAndQuery('ends-at').default(DEFAULT_ENDS_AT).toDate().isISO8601(),
];
```

## Expected behavior

Should add the default value for any missing param to both places instead of adding default only to `req.body`

## Current behavior

Adds default value only to `req.body`

## Express-validator version:

- Version: 6.10.0

**Comment by fedeci** [comment]
This seems not to be caused by the `default()` sanitizer, but by `buildCheckFunction()` that evaluates field results as an union and not as an intersection. When the context is run if at least one of the specified locations passes the validation then the whole `check` passes.
You can reproduce the "bug" also running:
```js
checkBodyAndQuery('id').isInt(),
```
and performing a request with empty query params and `{ id: 1 }` as body.
As you can see it passes validation. 
To avoid this, you should specify different validation chains:
```js
[
  body('id').isInt(), 
  query('id').isInt(),
]
```

**Comment by shizpi** [comment]
@fedeci Yes, I want it to pass the validation. It should be in query OR in body so the validation works as intended.

But when `starts-at` is missing in both places the default value is added only to req.body, instead of adding it also to `req.query`.

If I would validate explicitly `query('starts-at').default(DEFAULT_STARTS_AT).toDate().isISO8601(),` then the default would be added to `req.query`.
